File: src/main/java/frc/robot/SuperPIDControllerGroup.java
package frc.robot;

import java.util.Arrays;

public class SuperPIDControllerGroup {
    SuperPIDController[] controllers;

    public SuperPIDControllerGroup(SuperPIDController[] controllers) {
        this.controllers = controllers;
    }

    public void stopAll() {
        Arrays.stream(controllers)
            .forEach(SuperPIDController::stop);
    }

    public void resetAll() {
        Arrays.stream(controllers)
            .forEach(SuperPIDController::reset);
    }

    public boolean allOnTarget() {
        return Arrays.stream(controllers)
            .allMatch(SuperPIDController::isOnTarget);
    }

    public void maybeSetAllConstantsFromDash() {
        Arrays.stream(controllers)
            .forEach(SuperPIDController::maybeSetConstantsFromDash);
    }

    public void maybePutAllConstantsToDash() {
        Arrays.stream(controllers)
            .forEach(SuperPIDController::maybePutConstantsToDash);
    }

    public void executeAll() {
        Arrays.stream(controllers)
            .forEach(SuperPIDController::execute);
    }
}

File: src/main/java/frc/robot/SuperPIDControllerGroup.kt
package frc.robot

class SuperPIDControllerGroup(val controllers: List<SuperPIDController>) {
    constructor(vararg controllers: SuperPIDController) : this(controllers.asList())

    fun stopAll() = controllers.forEach(SuperPIDController::stop)
    fun resetAll() = controllers.forEach(SuperPIDController::reset)
    fun allOnTarget(): Boolean = controllers.all(SuperPIDController::isOnTarget)
    fun maybeSetAllConstantsFromDash() = controllers.forEach(SuperPIDController::maybeSetConstantsFromDash)
    fun maybePutAllConstantsToDash() = controllers.forEach(SuperPIDController::maybePutConstantsToDash)

    fun executeAll() = controllers.forEach(SuperPIDController::execute)
}